### PR TITLE
added support for extend a dataset split

### PR DESCRIPTION
**Pull request by de-code**
This is useful where we split the dataset previously and now have more data available. In that case it makes sense to keep the original split for the data we already had (e.g. we may have already trained a model on the training set and if we shouldn't use data for training that may not be in a test set).

It will also backup the previous lists.

### CODE REVIEW

File: sciencebeam_utils/tools/split_csv_dataset.py [hunk 1]
```diff
@@ -63,6 +69,55 @@ def split_rows(rows, percentages, fill=False):
     ]
 
 
+def _to_hashable(value):
+    return str(value)
```
Comment (GiancarloFusiello) [inline sciencebeam_utils/tools/split_csv_dataset.py]: Do you need this function?

File: sciencebeam_utils/tools/split_csv_dataset.py [hunk 2]
```diff
@@ -63,6 +69,55 @@ def split_rows(rows, percentages, fill=False):
...
+def _split_rows_without_existing_split(rows, percentages, fill=False):
+    chunk_size_list = get_chunk_size_list(len(rows), percentages, fill=fill)
+    return split_row_chunks(rows, chunk_size_list)
+
+
+def _substract_list(list1, list2):
+    return [
+        a - b
+        for a, b in zip(list1, list2)
+    ]
```
Comment (GiancarloFusiello) [inline sciencebeam_utils/tools/split_csv_dataset.py]: Will this logic fit on one line?

File: sciencebeam_utils/tools/split_csv_dataset.py (hunk 1 above)
Comment (de-code) [reply sciencebeam_utils/tools/split_csv_dataset.py]: I am adding `rows` to a set. A `row` will in reality be an array (but in tests may be a string or a number). I only use the set to check whether they exist, i.e. I don't need a reverse. Here calling `str` which will format an array as something like `["value1", "value2"]` seems to do. But I could call join, calculate a hash or do anything similar. The faster the better. But it's not used that often and doesn't need to be that optimised if it's fast enough.

Where I use the function I don't care about the details, which is why I have a separate function.

File: sciencebeam_utils/tools/split_csv_dataset.py (hunk 2 above)
Comment (de-code) [reply sciencebeam_utils/tools/split_csv_dataset.py]: I can move it on a single line.

File: sciencebeam_utils/tools/split_csv_dataset.py (hunk 1 above)
Comment (de-code) [reply sciencebeam_utils/tools/split_csv_dataset.py]: Okay, I've switched to using `tuple` and changed the tests to always pass in the rows with the correct type. I've also added type hints (comments for Python 2).
